### PR TITLE
chore: remove console errors on dev

### PR DESCRIPTION
**Pull request by rishabhrathod01**
## Description

This PR omits props to be sent to the DOM component as they are not required, leading to errors on development.

Fixes 

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11142003456>
> Commit: 49353f7be46bbb5ec9c581ed6a0adf975f65d32e
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11142003456&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Wed, 02 Oct 2024 10:58:14 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Release Notes

- **New Features**
  - Introduced `isMultiLine` prop for enhanced multi-line input handling.
  - Added new customization options for buttons, including `primaryColor`, `navColorStyle`, and more.
  - Enhanced `IconButtonComponent` with additional props for greater flexibility.

- **Bug Fixes**
  - Improved error handling for Google reCAPTCHA integration.
  - Corrected property name from `varient` to `variant` across various components to ensure consistent styling.

- **Chores**
  - Updated component structures to ensure unique identification during rendering.
  - Minor typographical corrections in comments for clarity.

### CODE REVIEW

File: app/client/src/widgets/ButtonWidget/component/index.tsx [hunk 1]
```diff
@@ -154,6 +154,11 @@ export const StyledButton = styled((props) => (
       "boxShadowColor",
       "buttonColor",
       "buttonVariant",
+      "primaryColor",
+      "navColorStyle",
+      "varient",
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/widgets/ButtonWidget/component/index.tsx:159]: @rishabhrathod01 ☝️

[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/widgets/ButtonWidget/component/index.tsx:159]: Is this something we can fix now? or does it take much more effort?

[the remark this answers is not in the document]
Comment (rishabhrathod01) [reply app/client/src/widgets/ButtonWidget/component/index.tsx:159]: @albinAppsmith included the fix to correct spelling of the prop in the component

**Comment by rishabhrathod01** [comment]
All cypress test has passed on EE for the same. https://github.com/appsmithorg/appsmith-ee/pull/5270

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**